pydicom/fileutil.py

```diff
@@ -126,6 +124,28 @@ def read_undefined_length_value(fp,
         If EOF is reached before delimiter found.
     """
     data_start = fp.tell()
+    defer_size = size_in_bytes(defer_size)
+
+    # It's common for an undefined length value item to be an
+    # encapsulated pixel data as defined in PS3.5 section A.4.
+    # Attempt to parse the data under that assumption, since the method
+    #  1. is proof against coincidental embedded sequence delimiter tags
+    #  2. avoids accumulating any data in memory if the element is large
+    #     enough to be deferred
+    #  3. does not double-accumulate data (in chunks and then joined)
+    #
+    # Unfortunately, some implementations deviate from the standard and the
+    # encapsulated pixel data-parsing algorithm fails. In that case, we fall
+    # back to a method of scanning the entire element value for the
+    # sequence delimiter, as was done historically.
+    if delimiter_tag == SequenceDelimiterTag:
+        was_value_found, value = _try_read_encapsulated_pixel_data(
```

Comment:
You could just return `value` and change the test to `if value is not None: return value`

Reply:
Thanks, @scaramallion, but the `True`/`False` indicator allows us to distinguish between a `None` value returned when we were unable to read the data, and one returned when we _were_ able to, but the value was too big so we deferred it, on [line 310](https://github.com/pydicom/pydicom/pull/1141/files#diff-83ea7c65d63ddf17b0cbba1ac97ea789R310).

I'm happy to take another approach if you like, such as introducing a different special case value to represent a deferred value (or an inability to read the encapsulated data).

Reply:
Ah, fair enough. I'm still working through it.